### ISSUE TITLE
Adding custom loss to multibox loss layer

### ISSUE THREAD

**Issue by ankitkumar8** [weiliu89/caffe#726]
Hi, how can I introduce custom loss to multibox layer (for example if i want newloss instead of softmax), I made the corresponding .cpp, .hpp and .cu file for new loss and also modified the caffe.proto txt file (procedure for standard caffe), now how can i get multibox layer to use that loss function. Acc to my unsderstanding, I need to modify multibox_loss_layer.cpp file but what files do I have to modify additionally.
Thank You.

**Comment by weiliu89** [comment]
You probably need to take a look at [here](https://github.com/weiliu89/caffe/blob/ssd/src/caffe/layers/multibox_loss_layer.cpp#L94-L128) and [here](https://github.com/weiliu89/caffe/blob/ssd/src/caffe/layers/multibox_loss_layer.cpp#L206-L239).

**Comment by ankitkumar8** [comment]
thank you, but i already modified those 2 still getting an error which says something like "MultiBoxLossParameter_ConfLossType doesn't have xyz loss" defined, where to add those definitions and if there are any additional files to be modified.

**Comment by weiliu89** [comment]
https://github.com/weiliu89/caffe/blob/ssd/src/caffe/proto/caffe.proto#L1254

**Comment by ankitkumar8** [comment]
Thank you :) this should probably do it!
